package/MDAnalysis/core/groups.py

```diff
@@ -850,6 +850,128 @@ def center_of_geometry(self, pbc=None, compound='group'):
+
+        Examples
+        --------
+
+        To find the total charge of a given :class:`AtomGroup`::
+
+            >>> sel = u.select_atoms('prop mass > 4.0')
+            >>> sel.accumulate('charges')
+
+        To find the total mass per residue of all CA :class:`Atoms<Atom>`::
+
+            >>> sel = u.select_atoms('name CA')
+            >>> sel.accumulate('masses', compound='residues')
+            
+        To find the maximal charge per fragment of a given :class:`AtomGroup`::
+        
+            >>> sel.accumulate('charges', compound="fragments", op=np.max)
+
+
+        .. versionadded:: 0.20.0
+        """
+        
+        atoms = self.atoms
+
+        try:
+            attribute_values = getattr(atoms, attribute)
+        except AttributeError:
+            raise NoDataError("Group does not have {} attribute.".format(attribute))
+        
+        comp = compound.lower()
+        
+        if comp == 'group':
+            return op(attribute_values)
+        elif comp == 'residues':
+            compound_indices = atoms.resindices
+        elif comp == 'segments':
+            compound_indices = atoms.segindices
+        elif comp == 'molecules':
+            try:
+                compound_indices = atoms.molnums
+            except AttributeError:
+                raise NoDataError("Cannot use compound='molecules': "
+                                  "No molecule information in topology.")
+        elif comp == 'fragments':
+            try:
+                compound_indices = atoms.fragindices
+            except NoDataError:
+                raise NoDataError("Cannot use compound='fragments': "
+                                  "No bond information in topology.")
+        else:
+            raise ValueError("Unrecognized compound definition: {}\nPlease use"
+                             " one of 'group', 'residues', 'segments', "
+                             "'molecules', or 'fragments'.".format(compound))
+        
+        # Sort attribute values by compound
+        sort_indices = np.argsort(compound_indices)
+        compound_indices = compound_indices[sort_indices]
+        
+        attribute_values = attribute_values[sort_indices]
+        # Get sizes of compounds:
+        unique_compound_indices, compound_sizes = np.unique(compound_indices,
+                                                            return_counts=True)
+        n_compounds = len(unique_compound_indices)
+        unique_compound_sizes = unique_int_1d(compound_sizes)
+        # Allocate output array:
+        accumulation = np.zeros(n_compounds)
+        # Compute sums per compound for each compound size:
+        for compound_size in unique_compound_sizes:
+            compound_mask = compound_sizes == compound_size
+            _compound_indices = unique_compound_indices[compound_mask]
+            atoms_mask = np.in1d(compound_indices, _compound_indices)
+            _summands = attribute_values[atoms_mask].reshape((-1, compound_size))
```

Comment:
What does "summands" mean as a variable name?

Reply:
These are the elements to which the accumulation is applied. I will rename them to elements if this is more clear.